craft_parts/executor/step_handler.py

```diff
@@ -173,13 +172,37 @@ def pkgconfig_fixup(file_path: str) -> None:
                 prefix_trim=self._part.part_install_dir,
             )
 
-        files, dirs = migrate_files(
-            files=files,
-            dirs=dirs,
-            srcdir=self._part.part_install_dir,
-            destdir=self._part.stage_dir,
-            fixup_func=pkgconfig_fixup,
-        )
+        if self._partitions:
```

Comment:
If using the `for_each_partition` idea this could be handled in a uniform way without exposed conditionals.